src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodCodeNode.cs

```diff
@@ -99,6 +99,26 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
                     dependencies.Add(new DependencyListEntry(factory.MethodEntrypoint(invokeStub), "Reflection invoke"));
             }
 
+            if (_method.HasInstantiation)
+            {
+                if (factory.MetadataManager.ExactMethodInstantiations.AddExactMethodInstantiationEntry(factory, _method))
+                {
+                    // Ensure dependency nodes used by the signature are added to the graph
+                    if (dependencies == null)
+                        dependencies = new DependencyList();
+
+                    dependencies.Add(new DependencyListEntry(factory.NecessaryTypeSymbol(_method.OwningType), "Exact method instantiation signature"));
+
+                    foreach (var arg in _method.Instantiation)
+                        dependencies.Add(new DependencyListEntry(factory.NecessaryTypeSymbol(arg), "Exact method instantiation signature"));
+
+                    dependencies.Add(new DependencyListEntry(factory.NecessaryTypeSymbol(_method.Signature.ReturnType), "Exact method instantiation signature"));
+
+                    for (int i = 0; i < _method.Signature.Length; i++)
+                        dependencies.Add(new DependencyListEntry(factory.NecessaryTypeSymbol(_method.Signature[i]), "Exact method instantiation signature"));
```

Comment:
So, I agree with Michal about not wanting to have the side effect of affecting the set of ExactMethodInstantiations, but I also want to make sure that these extra dependencies (which are actually conditional dependencies of MethodCodeNode and the ExactMethodInstantiationsNode) are tracked in a way that makes it obvious these are because of the combination of the two nodes, not just because of the MethodCodeNode.